### DISCUSSION TITLE
Running jdl on Windows 11 with NVIDIA 4070ti. Should I install Ubuntu and WSL?  Which cuda version?

### DISCUSSION THREAD

**Discussion by DonaldAlan** [deepjavalibrary/djl#2738] [Q&A] [answered]
I have Windows 11 with a NVIDIA 4070ti.     Neither TrainBertOnCode nor TrainResnetWithCifar10 runs on the gpu. I get:

[INFO ] - Ignore mismatching platform from: jar:file:/D:/.gradle/caches/modules-2/files-2.1/ai.djl.mxnet/mxnet-native-cu112mkl/1.9.1/3e0e468585ed841cdcc93d34908a21f55b515950/mxnet-native-cu112mkl-1.9.1-linux-x86_64.jar!/native/lib/mxnet.properties

My system has CUDA 12.1 (/c/Program\ Files/NVIDIA\ GPU\ Computing\ Toolkit/CUDA/v12.1/)  and a directory with  CUDA 11.6 (/d/ProgramFilesD/cuda-11.6.2/toolkit , as seen with git bash shell).    Changing paths in build.gradle with systemProperties, jdl still won't run examples on the gpu.        I presume that I need to downgrade  CUDA to 11.2 on my OS, but that doesn't seem easy to do.  

Anyway, would it be easier for me to install Ubuntu using WSL(2)?       Can I make the examples use something other than MXNet?             **Clearer instructions and recommended setups would help.**

**Comment by frankfliu** [comment, marked as answer]
MXNet doesn't support CUDA 12, it can only run on CUDA 11.2.

You can try to use PyTorch. PyTorch should be able to run with CUDA 12.1.

You can set system properties to use PyTorch: `System.setProperty("ai.djl.default_engine", "PyTorch")`

**Comment by DonaldAlan** [comment]
It works now. Thanks.

I installed CUDA 11.7 alongside several other CUDAs in   /c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/ (as seen from git bash).:

$ cd /c/Program Files/NVIDIA GPU Computing Toolkit/CUDA
$ ls
total 12
4 v11.6/  4 v11.7/  4 v12.1/

And in examples/build.gradle I added:

run {
   System.setProperty("ai.djl.default_engine", "PyTorch")
   ...
}

CUDA_PATH is set to C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7

**Comment by DonaldAlan** [comment]
It stopped working but I fixed it: I found that I also have to do

    export PYTORCH_VERSION=1.13.1  
    
Otherwise, it tries to use version 2.01. Both are in my .djl.ai/pytorch/ folder:

$ ls -l
total 28
4 drwxr-xr-x 1 Think 197609    0 Jul 29 15:34 1.13.1-cpu-win-x86_64/
4 drwxr-xr-x 1 Think 197609    0 Aug  5 13:44 1.13.1-cu117-win-x86_64/
8 -rw-r--r-- 1 Think 197609 4338 May 23 08:19 1.13.1.txt
4 drwxr-xr-x 1 Think 197609    0 Aug 24 08:50 2.0.1-cpu-win-x86_64/
8 -rw-r--r-- 1 Think 197609 4463 Aug 24 08:50 2.0.1.txt

BTW, this is on Windows 11, using git bash.
